jedi/parser/tree.py

```diff
@@ -52,6 +53,19 @@ def is_node(node, *symbol_names):
         return type in symbol_names
 
 
+def _fix_forward_reference(annotation):
+    if isinstance(annotation, String):
```

Comment:
Right indeed! It's also that the whole "when you see a string, you should covert that as a forward reference"-thing is a pep-0484 implementation, not general annotations. I'll give it a shot! Not too sure where to make the test, since `type_factory(list)` is probably not allowed in pep-0484 (even though no harm in supporting it), and something like `typing.Union[int, str]` is not something supported by the PR yet. Perhaps adding a `FIXTURE` to test_parser_tree.py ?